url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/util/mxLightweightLabel.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * $Id: mxLightweightLabel.java,v 1.1 2010-11-30 19:41:25 david Exp $
+ * Copyright (c) 2007, Gaudenz Alder
+ */
+package com.mxgraph.util;
+
+import java.awt.Font;
+import java.awt.Rectangle;
+
+import javax.swing.JLabel;
+import javax.swing.SwingConstants;
+
+/**
+ * @author Administrator
+ * 
+ */
+public class mxLightweightLabel extends JLabel
+{
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -6771477489533614010L;
+
+	/**
+	 * 
+	 */
+	protected static mxLightweightLabel sharedInstance;
+
+	/**
+	 * Initializes the shared instance.
+	 */
+	static
+	{
+		try
+		{
+			sharedInstance = new mxLightweightLabel();
+		}
+		catch (Exception e)
+		{
+			// ignore
+		}
+	}
+
+	/**
+	 * 
+	 */
+	public static mxLightweightLabel getSharedInstance()
+	{
+		return sharedInstance;
+	}
+
+	/**
+	 * 
+	 * 
+	 */
+	public mxLightweightLabel()
+	{
+		setFont(new Font(mxConstants.DEFAULT_FONTFAMILY, 0,
+				mxConstants.DEFAULT_FONTSIZE));
+		setVerticalAlignment(SwingConstants.TOP);
+	}
+
+	/**
+	 * Overridden for performance reasons.
+	 * 
+	 */
+	public void validate()
+	{
+	}
+
+	/**
+	 * Overridden for performance reasons.
+	 * 
+	 */
+	public void revalidate()
+	{
+	}
+
+	/**
+	 * Overridden for performance reasons.
+	 * 
+	 */
+	public void repaint(long tm, int x, int y, int width, int height)
+	{
+	}
+
+	/**
+	 * Overridden for performance reasons.
+	 * 
+	 */
+	public void repaint(Rectangle r)
+	{
+	}
+
+	/**
+	 * Overridden for performance reasons.
+	 * 
+	 */
+	protected void firePropertyChange(String propertyName, Object oldValue,
+			Object newValue)
+	{
+		// Strings get interned...
+		if (propertyName == "text")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.